test/upgrade/verify_jobs.go

```diff
@@ -26,6 +27,72 @@ func VerifyPostInstallJobs(cfg VerifyPostJobsConfig) upgrade.Operation {
 	})
 }
 
+func VerifyPostInstallServingJobs(ctx *test.Context, cfg VerifyPostJobsConfig) upgrade.Operation {
+	return upgrade.NewOperation("Verify jobs in "+cfg.Namespace, func(c upgrade.Context) {
+		if err := verifyPostInstallServingJobs(context.Background(), ctx, c, cfg); err != nil {
+			c.T.Error(err)
+		}
+	})
+}
+
+func verifyPostInstallServingJobs(ctx context.Context, testCtx *test.Context, c upgrade.Context, cfg VerifyPostJobsConfig) error {
+	jobs, err := testCtx.Clients.Kube.
```

Comment:
This function is completely same as `verifyPostInstallJobs`, except for initializing the clients. I think you could use the same approach as here for Eventing - get the clients from testCtx. Then this whole function could be reused and code duplication removed.

Reply:
Eventing uses `client := testlib.Setup(c.T, false)` that does a lot more that is needed in this case. It is not the same setup.

Reply:
And I think the `testlib.Setup` is not necessary because we only need the client to work with `Jobs` resources. We don't create any namespace here.

Reply:
Yeah ok I cleaned it up for Eventing too ;)

Reply:
> PASS test/upgrade.TestServerlessUpgrade/PostUpgradeTests/Verify_jobs_in_knative-serving (0.03s)
PASS test/upgrade.TestServerlessUpgrade/PostUpgradeTests/Verify_jobs_in_knative-eventing (607.77s)

That's huge :thinking:

Reply:
I think it is related to tear down logic. I hope unification to reduce time for Eventing too.

Reply:
Still too slow:
```
PASS test/upgrade.TestServerlessUpgrade/PostUpgradeTests/Verify_jobs_in_knative-serving (5.07s)
PASS test/upgrade.TestServerlessUpgrade/PostUpgradeTests/Verify_jobs_in_knative-eventing (620.03s)
```